src/bin/btc-cold.rs

```diff
@@ -421,11 +447,11 @@ impl Args {
             return Err(Error::DescriptorDerivePattern);
         }
         for index in skip..(skip + count) {
-            let address = DescrTrait::<bitcoin::PublicKey>::address(&descriptor, &secp, &[
+            let script = DescrTrait::<bitcoin::PublicKey>::script_pubkey(&descriptor, &secp, &[
```

Comment:
Why we can't use the previous method now? Unnwrapping is prohibited in the library code (except tests). This should be either expect, but much better to use `AddressCompat` specifically designed for this purpose.

Reply:
I'm not sure about your question.

I changed the method because the Descriptor doesn't save the network correctly.

When we use strict_decode in the Descriptor, it always returns Testnet instead of Regtest. See details [here](https://github.com/orgs/LNP-BP/discussions/128#discussioncomment-3629829)

Therefore, I needed to use a method that allows passing the network as a parameter.

Reply:
No, I meant to update `CompatAddr` to support regtest. If this is challenging I can work on a PR doing that

Reply:
Ahh Ok, I understand now.

I will make this change.